src/transaction/egld.ts

```diff
@@ -0,0 +1,992 @@
+    EsdtToken,
+    EsdtIssue,
+    EsdtTransfer,
+    EsdtMint,
+    EsdtFreezeOrWipeOrOwnership,
+    EsdtSpecialRole,
+    EsdtControlChanges,
+    EsdtIssueNftOrSft,
+    EsdtCreateNftOrSft,
+    EsdtTransferNftCreateRole,
+    EsdtAddOrBurnNftQuantity,
+    EsdtFreezeOrWipeNft,
+    EsdtTransferNft,
+    EgldEsdtTransaction,
+    Fee,
+    TransactionKMS,
+} from '../model';
+import {generateAddressFromPrivatekey} from '../wallet/address';
+
+const ELROND_V3_ENDPOINT = `${TATUM_API_URL}/v3/egld/web3/${process.env.TATUM_API_KEY}`;
+
+/**
+ * Get Elrond network config
+ */
+export const egldGetConfig = async () => {
+    const gasStationUrl = await getEgldClient('https://api.elrond.com'); // TODO: use TATUM API endpoint
+    try {
+        const {data} = await axios.get(`${gasStationUrl}/network/config`);
+        return data;
+    } catch (e) {
+    }
+    return null;
+};
+
+/**
+ * Estimate Gas price for the transaction.
+ */
+export const egldGetGasPrice = async (): Promise<number> => {
+    const config = await egldGetConfig();
+    return config?.erd_min_gas_price || 1000000000;
+};
+
+/**
+ * Estimate Gas limit for the transaction.
+ */
+const egldGetGasLimit = async (tx: EgldSendTransaction): Promise<number> => {
+    // TODO: use this as TATUM API endpoint
+    // const gasStationUrl = 'https://api.elrond.com';
+    // // const gasStationUrl = await getEgldClient();
+    // try {
+    //     const {data} = await axios.post(`${gasStationUrl}/transaction/cost`, tx);
+    //     return data?.txGasUnits || 50000;
+    // } catch (e) {
+    // }
+    // return 50000;
+
+    return new BigNumber((await egldEstimateGas(tx)) || 50000).toNumber();
+};
+
+/**
+ * Get account nonce
+ */
+const egldGetAccountNonce = async (from: string): Promise<number> => {
+    // TODO: use this as TATUM API endpoint
+    // const {data} = await axios.get(`${ELROND_V3_ENDPOINT}/address/${from}/nonce`);
+    // return data?.nonce;
+    return await egldGetTransactionsCount(from);
+};
+
+/**
+ * Sign transaction
+ */
+export const signEgldTransaction = async (tx: EgldSendTransaction, fromPrivateKey: string): Promise<string> => {
+    const message = new Uint8Array(Buffer.from(JSON.stringify(tx)));
+
+    const pair = tweetnacl.sign.keyPair.fromSeed(new Uint8Array(Buffer.from(fromPrivateKey, 'hex')));
+    const signingKey = pair.secretKey;
+    const signature = Buffer.from(tweetnacl.sign(message, signingKey)).toString('hex');
+    // "tweetnacl.sign()" returns the concatenated [signature, message], therfore we remove the appended message:
+    tx.signature = signature.slice(0, signature.length - message.length);
+
+    return JSON.stringify(tx);
+};
+
+/**
+ * Returns EGLD server to connect to.
+ * @param provider url of the EGLD Server to connect to. If not set, default public server will be used.
+ * @param fromPrivateKey optional private key of sender account
+ */
+export const getEgldClient = (provider?: string, fromPrivateKey?: string) => {
```

Comment:
if you just return string, private key doesn't have to be here.

Reply:
ok